main.cpp

```diff
@@ -24,408 +22,153 @@
  
  7) make sure it compiles & runs without errors.
  
- 8) use your knowledge of casting to remove any conversion warnings. 
+ 8) use your knowledge of casting to remove any conversion warnings.
  
  Commit your changes by clicking on the Source Control panel on the left, entering a message, and click [Commit and push].
  
 Make a pull request after you make your first commit and pin the pull request link to our DM thread.
-
 send me a DM to check your pull request
-
  Wait for my code review.
  */
 
-
-
-
 #include <iostream>
 
-struct DoubleType;
-struct IntType;
-
 struct FloatType
 {
-    FloatType(float val) : value(new float(val)) {}
-    ~FloatType() { delete value; }
+    FloatType(float val = 0.0f) : m_value(new float(val)) {}
+    ~FloatType()
+	{
+		delete m_value;
+		m_value = nullptr;
+	}
```

Comment:
your previous implementations were fine. 
- [ ] FIXME revert to previous implementation.

Reply:
I saw a reply on [this thread](https://stackoverflow.com/questions/1931126/is-it-good-practice-to-null-a-pointer-after-deleting-it) and again thought it was good practice, just like checking for `nullptr` before using it.